### PR TITLE
update validation_testing to use ECR

### PR DESCRIPTION
**Pull request by aimethed**
*Description of changes:*
Update the validation testing to deploy an ECR repository, build and push an image of the connector to the created repository, and update the template to point to the correct ImageUri.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aimethed** [comment]
Confirmed it works by running locally and through github: https://github.com/awslabs/aws-athena-query-federation/actions/runs/11466526706

File: validation_testing/deploy_infra.sh [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+aws ecr create-repository --repository-name athena-federation-repository-$CONNECTOR_NAME --region us-east-1
+
+# push the ECR image for the connector to the ECR repository
+aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin $ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com
+docker build -t athena-federation-repository-$CONNECTOR_NAME $REPOSITORY_ROOT/athena-$CONNECTOR_NAME
+docker tag athena-federation-repository-$CONNECTOR_NAME\:latest $ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/athena-federation-repository-$CONNECTOR_NAME\:latest
+docker push $ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/athena-federation-repository-$CONNECTOR_NAME\:latest
+
+# update the template to use the correct ImageUri
+sed -i "s|292517598671|$ACCOUNT_ID|g" "$REPOSITORY_ROOT/athena-$CONNECTOR_NAME/athena-$CONNECTOR_NAME.yaml"
```
Comment (AbdulR3hman) [inline validation_testing/deploy_infra.sh:18]: we're exposing an account number here; we can make this into a github secret

File: validation_testing/run_release_tests.sh [hunk 2]
```diff
@@ -7,9 +7,24 @@
...
+aws ecr create-repository --repository-name athena-federation-repository-$CONNECTOR_NAME --region us-east-1
+
+# push the ECR image for the connector to the ECR repository
+aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin $ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com
+docker build -t athena-federation-repository-$CONNECTOR_NAME $REPOSITORY_ROOT/athena-$CONNECTOR_NAME
+docker tag athena-federation-repository-$CONNECTOR_NAME\:latest $ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/athena-federation-repository-$CONNECTOR_NAME\:latest
+docker push $ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/athena-federation-repository-$CONNECTOR_NAME\:latest
+
+# update the template to use the correct ImageUri
+sed -i "s|292517598671|$ACCOUNT_ID|g" "$REPOSITORY_ROOT/athena-$CONNECTOR_NAME/athena-$CONNECTOR_NAME.yaml"
```
Comment (AbdulR3hman) [inline validation_testing/run_release_tests.sh:24]: same here

File: validation_testing/deploy_infra.sh (hunk 1 above)
Comment (aimethed) [reply validation_testing/deploy_infra.sh:18]: That account number is already in all of our yaml templates. It's something I brought up with AppSec when initially discussing ECR deployment and they said its fine.

Comment (aimethed) [reply validation_testing/deploy_infra.sh:18]: This goes and replaces the default account ID (the one already exposed in yaml templates) with the secret one used by github actions (gotten from the credentials so never exposed)

**Review by AbdulR3hman — APPROVED** [review]

**Pull request merged**